### PR TITLE
Add ega dbgap

### PR DESCRIPTION
**Pull request by jacobwindsor**
dcp-418

### CODE REVIEW

File: src/app/projects/components/filters/filters.component.ts [hunk 1]
```diff
@@ -24,7 +25,7 @@ export class FiltersComponent implements OnInit {
   @Output()
   valueSearched = new EventEmitter<string>();
 
-  dataLocations = ['HCA Data Portal', 'GEO', 'ENA', 'ArrayExpress', 'EGA'];
+  dataLocations = ProjectsService.allowedLocations;
```
Comment (amnonkhen) [inline src/app/projects/components/filters/filters.component.ts:28]: 👍

File: src/app/projects/pages/projects-list/projects-list.component.ts [hunk 2]
```diff
@@ -113,6 +113,7 @@ export class ProjectsListComponent implements OnInit, OnDestroy {
       arrayExpressAccessions: 'Arrayexpress',
       geoAccessions: 'GEO',
       egaStudiesAccessions: 'EGA',
+      dbgapAccessions: 'dbGaP',
```
Comment (amnonkhen) [inline src/app/projects/pages/projects-list/projects-list.component.ts:116]: Did you consider reusing the names defined in the ProjectService?

File: src/app/projects/services/projects.service.ts [hunk 3]
```diff
@@ -152,27 +168,27 @@ export class ProjectsService implements OnDestroy {
       return false;
     }
     switch (filters.location) {
-      case 'HCA Data Portal':
+      case ProjectsService.allowedLocations[0]:
```
Comment (amnonkhen) [inline src/app/projects/services/projects.service.ts:171]: I wouuld suggest to use an enum or a dictionary with named keys to increase readability.

File: src/app/summary/services/projects.data.ts [hunk 4]
```diff
@@ -1,31 +1,42 @@
...
+    title: '',
+    uuid: '',
     organs: [],
     technologies: [],
-    cellCount: 0
+    cellCount: 0,
   };
 }
 
 export function makeDummyProjects() {
```
Comment (amnonkhen) [inline src/app/summary/services/projects.data.ts:24]: Is this for testing? What does "dummy" mean here? If it is for testing, it should be moved to a testing file.

**Review by amnonkhen — CHANGES_REQUESTED** [review]

File: src/app/projects/services/projects.service.ts [hunk 5]
```diff
@@ -15,6 +15,15 @@ interface Filters {
 
 @Injectable()
 export class ProjectsService implements OnDestroy {
+  static allowedLocations = [
```
Comment (yusra-haider) [inline src/app/projects/services/projects.service.ts:18]: just curious -- why did you move these into the service from the component?

**Review by yusra-haider — DISMISSED** [review]

File: src/app/projects/services/projects.service.ts [hunk 6]
```diff
@@ -125,6 +134,13 @@ export class ProjectsService implements OnDestroy {
   }
 
   setFilters(filters: Filters) {
+    if (
+      filters.location &&
+      !ProjectsService.allowedLocations.includes(filters.location)
+    ) {
+      console.error(`${filters.location} not an allowed data location.`);
```
Comment (ke4) [inline src/app/projects/services/projects.service.ts:141]: Is that right that we only log the error to the console?

**Review by ke4 — CHANGES_REQUESTED** [review]
I added a small comment. Could you consider it, please? Tx

File: src/app/summary/services/projects.data.ts (hunk 4 above)
Comment (jacobwindsor) [reply src/app/summary/services/projects.data.ts:24]: I did not write this, but I assume it means mock

File: src/app/projects/services/projects.service.ts (hunk 6 above)
Comment (jacobwindsor) [reply src/app/projects/services/projects.service.ts:141]: Yes, I dont think this should be a fatal error that blocks the whole app

File: src/app/projects/services/projects.service.ts (hunk 5 above)
Comment (jacobwindsor) [reply src/app/projects/services/projects.service.ts:18]: Because the service does the searching too and knows what things are searchable so it acts as a source of truth for projects

File: src/app/projects/services/projects.service.ts (hunk 6 above)
Comment (amnonkhen) [reply src/app/projects/services/projects.service.ts]: I think we should be reporting app errors to a central location, like google analytics or Grafana, in order to be able to learn about app errors in the wild.

File: src/app/projects/pages/projects-list/projects-list.component.ts (hunk 2 above)
Comment (jacobwindsor) [reply src/app/projects/pages/projects-list/projects-list.component.ts:116]: I think it's better not to reuse here. The names defined in the ProjectService are allowed data locations and are to do with the UI and may not necessarily be the same as spreadsheet output (in fact they're not, spreadsheet only shows EGAS, not EGAD)

File: src/app/summary/services/projects.data.ts (hunk 4 above)
Comment (amnonkhen) [reply src/app/summary/services/projects.data.ts:24]: Let's leave it for now.

**Review by amnonkhen — APPROVED** [review]

**Review by amnonkhen — APPROVED** [review]

**Review by yusra-haider — APPROVED** [review]

**Review by ke4 — APPROVED** [review]
